src/GrafanaDashboardDeploy/src/GrafanaClient.cs

```diff
@@ -0,0 +1,327 @@
+            }
+            else
+            {
+                _logger.LogWarning("Response does not indicate success");
+                _logger.LogWarning("Status {}: {}", response.StatusCode, response.ReasonPhrase);
+                _logger.LogDebug(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+            }
+
+            return folder;
+        }
+
+        /// <summary>
+        /// Get a Grafana Folder by its id
+        /// </summary>
+        /// <param name="id">The folder id</param>
+        /// <returns>The Folder JSON object</returns>
+        public async Task<JObject> GetFolder(int id)
+        {
+            var uri = new Uri(new Uri(BaseUrl), $"/api/folders/id/{id}");
+            var folder = new JObject();
+
+            HttpResponseMessage response = await _client.GetAsync(uri).ConfigureAwait(false);
+
+            if (response.IsSuccessStatusCode)
+            {
+
+                using (Stream stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false))
+                using (var streamReader = new StreamReader(stream))
+                using (var jsonReader = new JsonTextReader(streamReader))
+                {
+                    folder = await JObject.LoadAsync(jsonReader).ConfigureAwait(false);
+                }
+            }
+            else
+            {
+                _logger.LogWarning("Response does not indicate success");
+                _logger.LogWarning("Status {}: {}", response.StatusCode, response.ReasonPhrase);
+                _logger.LogDebug(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+            }
+
+            return folder;
+        }
+
+        /// <summary>
+        /// Get a Data Source by its name
+        /// </summary>
+        /// <param name="name">The data source name</param>
+        /// <returns>The Data Source JSON object as defined by the Grafana Data Source API</returns>
+        public async Task<JObject> GetDataSource(string name)
+        {
+            var uri = new Uri(new Uri(BaseUrl), $"/api/datasources/name/{name}");
+            var folder = new JObject();
+
+            HttpResponseMessage response = await _client.GetAsync(uri).ConfigureAwait(false);
+
+            if (response.IsSuccessStatusCode)
+            {
+
+                using (Stream stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false))
+                using (var streamReader = new StreamReader(stream))
+                using (var jsonReader = new JsonTextReader(streamReader))
+                {
+                    folder = await JObject.LoadAsync(jsonReader).ConfigureAwait(false);
+                }
+            }
+            else
+            {
+                _logger.LogWarning("Response does not indicate success");
+                _logger.LogWarning("Status {}: {}", response.StatusCode, response.ReasonPhrase);
+                _logger.LogDebug(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
```

Comment:
Are the errors from grafana in a format we can extract some human readable error message we can show to the user?